lib/wasix/src/runtime/package_loader/builtin_loader.rs

```diff
@@ -419,13 +423,17 @@ pub enum CacheValidationMode {
 // FIXME: This implementation will block the async runtime and should use
 // some sort of spawn_blocking() call to run it in the background.
 #[derive(Debug)]
-struct FileSystemCache {
+pub struct FileSystemCache {
     cache_dir: PathBuf,
 }
 
 impl FileSystemCache {
     const FILE_SUFFIX: &'static str = ".bin";
 
+    fn temp_dir(&self) -> PathBuf {
```

Comment:
I am not familiar with that part of the codebase, but won't using a shared  `__temp__` lead to issues, when multiple wasmer processes are unpacking packages at the same time? (edited